themes/default/content/what-is/what-are-docker-configs.md

```diff
@@ -0,0 +1,193 @@
+    - items:
+      - mindbody
+      - sourcegraph
+      - fenergo
+      - skai
+      - lemonade
+    - items:
+      - clearsale
+      - angellist
+      - webflow
+      - supabase
+      - ro
+---
+
+Docker, a leading platform in containerization technology, has revolutionized how applications are developed, shipped, and deployed. An essential facet of this ecosystem is the effective management of configuration data. [Docker Configs](https://docs.docker.com/engine/swarm/configs/) is a feature specially crafted to handle non-sensitive configuration information within Docker environments. This guide explores the ins and outs of Docker Secrets, highlighting its importance, functionality, and best practices.
+
+### What are Docker Configs?
+
+Docker Configs are a resource in Docker for storing non-sensitive information such as configuration files, separate from a service's image or running containers. This enables keeping Docker images as generic as possible without the need for bind-mounting configuration files into containers or using environment variables. Unlike Docker Secrets, Docker Configs are not encrypted at rest and are directly mounted into the container's filesystem.
+
+#### Key Features
+
+- **Separation of Configuration from Code**: Docker Configs allow you to store configuration files outside of your Docker images, leading to more generic and reusable images.
+- **Flexibility in Management**: Configs can be added, updated, or removed from services dynamically, without the need to rebuild or restart containers.
+- **Secure Transmission**: Configs are sent to the swarm manager over a mutual TLS connection and are stored securely.
+- **Easy Access within Containers**: Configs are automatically mounted into the container's filesystem, making them easily accessible to applications.
+- **Support for Various Data Types**: Configs can store generic strings or binary content, providing flexibility for different types of configuration data.
+
+### Creating Configs
+
+Docker Configs can be created via the Docker CLI.
+
+#### Prerequisites
+
+##### Install Docker
+
+Before creating configs in Docker, you must first make sure you have [Docker installed](https://docs.docker.com/get-docker/). Once you have installed Docker, enable and start the Docker service.
+
+```bash
+sudo systemctl enable docker
+sudo systemctl start docker
+```
+
+You can optionally add your user to the Docker group to provide non-root access.
+
+```bash
+sudo usermod -a -G docker ${USER}
+```
+
+Restart your terminal to apply the changes to the group, then check that Docker has installed correctly.
+
+```bash
+docker --version
+```
+
+{{< notes type="info" >}}
+
+The command to start Docker depends on your operating system. The above commands show examples for how to do this on Linux. You can find the commands relevant to your own operating system in the [Docker documentation](https://docs.docker.com/engine/install/).
+
+{{< /notes >}}
+
+##### Create a Swarm
```

Comment:
it seems like configs only matter with swarm but in the earlier paragraphs you never mention that fact. something to maybe elaborate

Reply:
It is included in the "prerequisites" section when talking about initializing a swarm and also in the considerations/challenges section.